filebeat/harvester/log.go

```diff
@@ -271,6 +271,10 @@ func (h *Harvester) close() {
 	harvesterClosed.Add(1)
 }
 
+// newLogFileReader creates a new reader to read log files
+// It creates a chain of readers which looks as following:
+// limit -> (multiline -> timeout) -> strip_newline -> json -> encode -> log_file
```

Comment:
I picked the other order an purpose as it is the caller / creation order and in case of an error, it goes back to the caller (left). But both orders work for me. I will update the text to make this even more clear. I see it as really important that this logic is understood when working with the code.